### PR TITLE
Add reverse dns enrichment fields to config_template.yaml.

### PR DESCRIPTION
**Pull request by jmw51798**
### What does this PR do?
Add reverse dns enrichment fields to config_template.yaml.

### CODE REVIEW

File: pkg/config/config_template.yaml
Comment (pgimalac) [inline pkg/config/config_template.yaml:4134]: I can't find those four configs declared or used anywhere in the agent repo ?

File: pkg/config/config_template.yaml
Comment (pgimalac) [inline pkg/config/config_template.yaml:4105]: I can't find those two configs declared or used anywhere in the agent repo ?

File: pkg/config/config_template.yaml
Comment (zoedt) [reply pkg/config/config_template.yaml:4134]: these are related to this PR that will be merged before this one!: https://github.com/DataDog/datadog-agent/pull/27876

File: pkg/config/config_template.yaml
Comment (zoedt) [inline pkg/config/config_template.yaml:4129]: small suggestion?: i don't know if there's a limit to how long the description would be - but maybe it would be worth giving an example with what this would look like for additional clarity? (or maybe documentation folks lmk if this already makes sense them!)

i.e. 
every 5 seconds (recovery interval), slowly ramp up over 5 (recovery intervals) back to 1000 (limit per seconds)

not sure if the anyone has a better way to put this than me, but just thought it's hard to grasp initially with so many configs and the name `recovery_interval` close to this one 😅

**Review by zoedt — APPROVED** [review]
lgtm once the other PR is merged :^)

File: pkg/config/config_template.yaml
Comment (jhgilbert) [inline pkg/config/config_template.yaml:4070]:
```suggestion
  ## The number of concurrent workers used to perform reverse DNS lookups.
```

File: pkg/config/config_template.yaml
Comment (jhgilbert) [inline pkg/config/config_template.yaml:4074]:
```suggestion
  ## The size of the channel used to send reverse DNS lookup requests to the workers.
```

File: pkg/config/config_template.yaml
Comment (jhgilbert) [inline pkg/config/config_template.yaml:4095]:
```suggestion
    ## An interval that specifies how often the cache is persisted to disk so the cache can be reloaded when the Agent is upgraded or restarted.
```

**Review by jhgilbert — APPROVED** [review]
Approved with a few suggestions, thank you!

File: pkg/config/config_template.yaml
Comment (jmw51798) [reply pkg/config/config_template.yaml]: Added more details and an example 32b7ba27c26032b39ad63758063e84f18fc1d558

**Pull request merged**
